SplitFiles/bread.c

```diff
@@ -31,12 +32,176 @@ BreadType*  rand_bread_type(BreadBin* bin) {
             }
         }
     }
+    //This should never be reached
+    printf("Error in rand_bread_type");
+    return rand_bread_type(bin);
+}
+
+BreadType* get_bread_type(BreadBin* bin, unsigned char typeLetter) {
+    if (typeLetter < 'A' || typeLetter > 'Z') {
+        return NULL;
+    }
+    int index = typeLetter - (unsigned char)'A';
+    return bin->breadTypes[index];
+}
+
+BreadState* get_bread(BreadBin* bin, unsigned char typeLetter) {
+    //int index = typeLetter - (unsigned char)'A';
+    //BreadType* type = *(bin->breadTypes + index);
+    BreadType* type = get_bread_type(bin, typeLetter);
+
+    if (type == NULL) {
+        return (BreadState *)NULL;
+    }
+    BreadState* newBread = malloc(sizeof(BreadState));
+    newBread->temperature = 0;
+    newBread->moisture = type->moisture + (rand() % type->moistureVariability);
+    newBread->toastedness = 0;
+    newBread->thermalMass = type->thermalMass + (rand() % type->massVariability);
+    newBread->thermalAggregation = 0;
+    newBread->type = type;
+    return newBread;
+}
+
+void add_bread(BreadBin* bin, BreadType* newType) {
+    //Set up in the bin
+    int index = newType->letter - (unsigned char)'A';
+    bin->breadTypes[index] = newType;
+    bin->probTotal += newType->callProb;
 }
 
 BreadBin* get_bread_bin() {
     BreadType** types = malloc(sizeof(BreadType*) * BREADBINSIZE);
+    for (int i = 0; i < BREADBINSIZE; ++i) {
+        types[i] = NULL;
+    }
     BreadBin* bin = malloc(sizeof(BreadBin));
     bin->probTotal = 0;
     bin->breadTypes = types;
 
-}
\ No newline at end of file
+    /*******************************************************
```

Comment:
Create all the bread types we will use - this is the only reference to the individual bread types in the code, everything else is generic based on the data provided here